### ISSUE TITLE
Mingw build error for dll files

### ISSUE THREAD

**Issue by barsaskn** [pjsip/pjproject#3404]
### Describe the bug

I'm trying to compile pjsip for windows, using mingw on linux. I configure with the --enable-shared option for DLL files, but I cannot compile because of the error I get in the make part.

### Steps to reproduce

1- Configure:

`./configure --host=x86_64-w64-mingw32 && make dep -j12 && make -j12`

2- Config_site.h

```
#define PJ_EXPORT_SPECIFIER  __declspec(dllexport
#define PJ_IMPORT_SPECIFIER  __declspec(dllimport)
#define PJ_DLL  1
#ifdef _LIB
#define PJ_EXPORTING 1
#endif
#include <pj/config_site_sample.h>
```

Also tried build on Windows using Visual Studio 2022 and Visual Studio 2019 with this config but only .lib files are created. 

3-Make: 

`make dep && make`

`make dep` is fine. But throws error in make part.

### PJSIP version

2.13

### Context

The issue happens on linux mingw

### Log, call stack, etc

```shell
/usr/bin/x86_64-w64-mingw32-ld: output/pjlib-x86_64-w64-mingw32/os_time_common.o:os_time_common.c:(.text+0x50): multiple definition of `pj_time_local_to_gmt'; output/pjlib-x86_64-w64-mingw32/os_time_win32.o:os_time_win32.c:(.text+0x210): first defined here
/usr/bin/x86_64-w64-mingw32-ld: output/pjlib-x86_64-w64-mingw32/os_time_common.o:os_time_common.c:(.text+0x70): multiple definition of `pj_time_gmt_to_local'; output/pjlib-x86_64-w64-mingw32/os_time_win32.o:os_time_win32.c:(.text+0x280): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [/home/bursiyer/Documents/pjproject/build/rules.mak:113: ../lib/libpj.dll.2] Error 1
make[2]: Leaving directory '/home/bursiyer/Documents/pjproject/pjlib/build'
make[1]: *** [Makefile:106: libpj.dll] Error 2
make[1]: Leaving directory '/home/bursiyer/Documents/pjproject/pjlib/build'
```

**Comment by PanDamien26** [comment]
Logs : https://pastebin.com/PDnNNmmH

**Comment by trengginas** [comment]
If you still having issue, try `make clean` or `make realclean` and repeat the build process.

**Comment by barsaskn** [comment]
> If you still having issue, try `make clean` or `make realclean` and repeat the build process.

Stilling getting same error while try to configure with --enable-shared with same error log.

`./configure --enable-shared --host=x86_64-w64-mingw32 && make dep -j12 && make -j12 `

**Comment by PanDamien26** [comment]
> If you still having issue, try `make clean` or `make realclean` and repeat the build process.

I redownloaded the last master. 
Followed what [barsaskn] did but i only see .exe and no .dll
What am i missing ?
Could someone send me a working .dll ?

**Comment by trengginas** [comment]
I manage to build the dll after the fix on #3408 (make sure you have this fix on the build).
Here are the steps:
- setup the build tools (mingw-64) as specified in #2598
- follow the instructions in https://docs.pjsip.org/en/latest/api/generated/pjlib/group/group__pj__dll__target.html#group-pj-dll-target
  - config_site.h 
    ```
    #define pj_export_specifier  __declspec(dllexport
    #define pj_import_specifier  __declspec(dllimport)
    #define pj_dll  1
    #define pj_exporting 1
    ```
  - `./configure --enabled-shared`
  - `make dep && make clean && make`
